Photocopying secret documents actually works now

optimumtact opened the pull request:
Bandit's photocopier changes rescued from a bad merge.
Closes #17399

Commit:
secret document photocopy actually works now

Commit:
Photocopying secret documents actually works now.

Commit:
makes requested changes, ends this endless meme

Core0verload commented on code/modules/paperwork/photocopier.dm:
```diff
@@ -127,7 +127,7 @@
 			for(var/i = 0, i < copies, i++)
 				if(toner > 5 && !busy && doccopy)
-					new /obj/item/documents/photocopy(src, doccopy)
+					new /obj/item/documents/photocopy(loc, src)
```
`doccopy` was kinda important

optimumtact committed:
correctly forward on doccopy argument

appease the cranky russian

phil235 commented on code/modules/paperwork/photocopier.dm:
```diff
@@ -245,6 +236,15 @@
 	flick("photocopier1", src)
 	updateUsrDialog()
 
+/obj/machinery/photocopier/proc/do_removal(obj/item/O, mob/user)
```
not the greatest proc name, but whatever

optimumtact replied:
I can refactor it to something more usefully named

phil235 commented on code/modules/paperwork/photocopier.dm:
```diff
@@ -245,6 +236,15 @@
+	else
+		O.loc = src.loc
+	user << "<span class='notice'>You take [O] out of [src].</span>"
+	updateUsrDialog()
```
Are you sure this updates the window correctly? Since you nullifies copy/photocopy/doccopy after doing the window update instead of before.

optimumtact replied:
I'll take a look and see

optimumtact committed:
Rename proc, move updateUsrDialog call

Ensures the dialog is called after the photocopy/doccopy is removed and
so the interface updates correctly

optimumtact commented:
Should be ready to go now
@tgstation/commit-access
